engine/src/main/java/com/datatorrent/stram/stream/BufferServerPublisher.java

```diff
@@ -184,6 +184,10 @@ public void setup(StreamContext context)
       statefulSerde = ((StatefulStreamCodec<Object>)StreamContext.CODEC.defaultValue).newInstance();
     } else if (codec instanceof StatefulStreamCodec) {
       statefulSerde = ((StatefulStreamCodec<Object>)codec).newInstance();
+      if (!codec.getClass().isAssignableFrom(statefulSerde.getClass())) {
```

Comment:
Why is the check necessary if validation already checks for the correct implementation of newInstance()?